runtime/src/bank/partitioned_epoch_rewards/calculation.rs

```diff
@@ -426,15 +426,23 @@ impl Bank {
 
                         let post_balance = stake_account.lamports();
                         total_stake_rewards.fetch_add(stakers_reward, Relaxed);
-                        return Some(StakeReward {
+                        let assert_msg = "solana_stake_program::rewards::redeem_rewards output \
+                                          is_ok only when a stake account has state \
+                                          StakeStateV2::Stake";
+                        let StakeStateV2::Stake(_meta, stake, _stake_flags) =
+                            stake_account.state().expect(assert_msg)
+                        else {
+                            panic!("{assert_msg}");
+                        };
```

Comment:
Yeah, these could totally just be an `unwraps`, if we don't care about the message.

I also thought about updating `redeem_rewards()` so that not only `stake_account` but `stake_state` is also mutable and updated inside the fn, to avoid the extra deserialization. But I didn't want to get that invasive in this PR.

> But man this reward redemption code path could definitely use a refactor someday to lean into type safety rather than so many unwraps

💯

Reply:
I went ahead and pushed a simplification. Don't feel too strongly.